### PR TITLE
Fix issues discussed in http://scala-exercises.47deg.com/koans#higherorderfunctions

### PR DESCRIPTION
**Pull request by mikela**
This pull request consists of two parts:
1. Fix typos in README.md and index.html.
2. Allow adding regular expressions as alternative solutions so that we can accept correct solutions not permitted now (eg, allowing "List(5,6,7)" besides "List(5, 6, 7)").

This is my first pull request so beware of dragons :P

### CODE REVIEW

File: app/json/higherorderfunctions.json [hunk 1]
```diff
@@ -75,6 +75,11 @@
         "List(\"ABC\", \"XYZ\", \"123\")",
         "List(\"abc\", \"xyz\", \"123\")",
         "List(5, 6, 7)"
+      ],      
+      "alternatives": [
+        "^List\\( *\"ABC\" *, *\"XYZ\" *, *\"123\" *\\)$", 
+        "^List\\( *\"abc\" *, *\"xyz\" *, *\"123\" *\\)$", 
+        "^List\\( *5 *, *6 *, *7 *\\)$",
```
Comment (rafaparadela) [inline app/json/higherorderfunctions.json]: The last comma makes fail when parsing

File: app/json/higherorderfunctions.json [hunk 2]
```diff
@@ -94,6 +99,10 @@
       "solutions": [
         "List(12, 20, 2)",
         "List(12, 20, 2)"
+      ],      
+      "alternatives": [
+        "^List\\( *12 *, *20 *, *2 *\\)$",
+        "^List\\( *12 *, *20 *, *2 *\\)$",
```
Comment (rafaparadela) [inline app/json/higherorderfunctions.json]: The last comma makes fail when parsing

**Comment by rafaparadela** [comment]
The JSON is not valid, other than that, the PR is interesting. Very useful for non-strict evaluations. Thank you very much.

**Comment by mikela** [comment]
Ah my bad! Fixed it right away (and validated with JSONLint).

**Comment by rafaparadela** [comment]
Cool! Shipping the boat...

**Pull request merged**